plugins/sqlfluff-templater-dbt/test/templater_test.py

```diff
@@ -490,8 +494,24 @@ def test__templater_dbt_handle_exceptions(
     finally:
         shutil.move(target_fpath, src_fpath)
         get_adapter(dbt_templater.dbt_config).connections.release()
+
+    # Debug logging.
+    print("Raised:", excinfo.value)
+    for trace in excinfo.traceback:
+        print(trace)
+
     # NB: Replace slashes to deal with different platform paths being returned.
-    assert exception_msg in excinfo.value.desc().replace("\\", "/")
+    if exception_class is SQLTemplaterError:
+        _msg = excinfo.value.desc().replace("\\", "/")
+    else:
+        _msg = str(excinfo.value).replace("\\", "/")
+    assert exception_msg in _msg
+    # Ensure that there's no context parent exception, because they don't pickle well.
+    # https://github.com/sqlfluff/sqlfluff/issues/6037
+    # We *should* be stripping any inherited exceptions from anything returned here.
+    # Any residual dbt exceptions are a risk for pickling errors.
+    assert not excinfo.value.__context__
```

Comment:
To make the test more robust and direct, could we have the test actually pickle and unpickle the exception to verify it doesn't crash?

Reply:
Ohh that's a great idea 💯